src/components/activity-completion/completion-page-content.tsx

```diff
@@ -1,50 +1,165 @@
-    onPageChange(0);
+    if (sequence) { onShowSequence?.(); }
+    else { onPageChange(0); }
+  };
+  const handleNextActivity = () => {
+    onActivityChange?.(activityNum + 1);
+    onPageChange(0); //TODO: This should go to the student's saved state page for the next activity
   };
   const handleShowAnswers = () => {
     showReport();
   };
-  const quotedActivityName = "\"" + activityName + "\"";
-  const completionText = `${quotedActivityName} activity complete!`;
-  const progressText = isActivityComplete
-      ? `Congratulations! You have reached the end of ${quotedActivityName} and your work has been saved`
-      : `It looks like you haven't quite finished ${quotedActivityName} yet. The answers you've given have been saved.`;
-  const nextStepText = isActivityComplete
-      ? `You have completed ${quotedActivityName} and you may exit now.`
-      : `You haven't completed ${quotedActivityName} yet. You can go back to complete it, or you can exit.`;
+
+  const sequenceProgress = () => {
+    let activityStatus;
+    sequence?.activities.map((sequenceActivity)=>{
+      activityStatus = activityProgress(sequenceActivity);    
+      const sequenceActivityComplete = activityStatus.numAnswers === activityStatus.numQuestions;
+      const activityCompletionArrayCopy = [...activityCompletionArray];
+      activityCompletionArrayCopy?.push(sequenceActivityComplete);
+      activityCompletionArray = [...activityCompletionArrayCopy];
+    });             
+    if (activityCompletionArray?.length === sequence?.activities.length && !activityCompletionArray?.includes(false)) {
+      return true;
+    } else {
+      return false;
+    }
+  };
+
+  const activityProgress = (kActivity: Activity) => {
+    let answerNum = 0, questionNum = 0, questionId: string;
+    let i = 0;
+    for (i = 0; i < kActivity.pages.length; i++) {
+      kActivity.pages[i].embeddables.map((embeddableWrapper: EmbeddableWrapper) => {
+        if (isQuestion(embeddableWrapper)) {
+          questionNum++;
+          questionId = refIdToAnswersQuestionId(embeddableWrapper.embeddable.ref_id);
+          answers?.map((answer: any) => {
+            //This does not take into account if user erase their open text response or answerTest is empty string 
+            // after response has been saved
+            if (answer.meta.question_id === questionId) { 
+              answerNum++;
+            }
+          });
+        }
+      });
+    }
+    return ({ numAnswers: answerNum, numQuestions: questionNum });
+  };
+
+  useEffect(() => {
+    watchAllAnswers(answerMetas => {
+      setAnswers(answerMetas);
+    });
+  },[]);
+
+  if(!answers) {
+    return (
+      <div className="completion-page-content" data-cy="completion-page-content">
+        <div className="progress-container" data-cy="progress-container">
+          <div className="progress-text">
+            Fetching your data ...
+          </div>
+        </div>
+      </div>  
+    );
+  }
```

Comment:
It's a little odd to have these two returns at different parts in the function.  I would try to use a single return with a ternary and even consider putting this piece above in its own small functional component to clean things up